### DISCUSSION TITLE
Latex package `siunitx` not identified by pandoc, even though it is installed

### DISCUSSION THREAD

**Discussion by lamyergeier** [jgm/pandoc#9642] [Q&A] [answered]
# Info

OS: Fedora 39

```
$ pandoc --version
pandoc 3.1.12.2
```

# siunitx package v3 is installed 

  ``` {.bash .numberLines}
  $ kpsepath tex | tr ':' '\n' \
    | perl -ne 'm-^(!!)?(/.*?)(//)?$- && print "$2\n"' \
    | while read path; do locate $path ; done | sort -u \
  ```

  ``` {.text .numberLines}
  /usr/share/texlive/texmf-dist/tex/generic/tex4ht/siunitx.4ht
  /usr/share/texlive/texmf-dist/tex/latex/lwarp/lwarp-common-mathjax-siunitx.sty
  /usr/share/texlive/texmf-dist/tex/latex/lwarp/lwarp-lltjp-siunitx.sty
  /usr/share/texlive/texmf-dist/tex/latex/lwarp/lwarp-siunitx-v2.sty
  /usr/share/texlive/texmf-dist/tex/latex/lwarp/lwarp-siunitx.sty
  /usr/share/texlive/texmf-dist/tex/latex/plautopatch/plsiunitx.sty
  /usr/share/texlive/texmf-dist/tex/latex/siunitx
  /usr/share/texlive/texmf-dist/tex/latex/siunitx/siunitx-abbreviations.cfg
  /usr/share/texlive/texmf-dist/tex/latex/siunitx/siunitx-binary.cfg
  /usr/share/texlive/texmf-dist/tex/latex/siunitx/siunitx-v2.sty
  /usr/share/texlive/texmf-dist/tex/latex/siunitx/siunitx-version-1.cfg
  /usr/share/texlive/texmf-dist/tex/latex/siunitx/siunitx.sty
  /usr/share/texlive/texmf-dist/tex/luatex/luatexja/patches/lltjp-siunitx.sty
  ```
# Unable to produce pdf

I did the following: 

```
$ printf 'Thus, total size is $4288+4=\qty{4292}{kiB}$' | pandoc -s --from=markdown+tex_m
ath_dollars+raw_tex --to latex -o a.pdf
Error producing PDF.
! Undefined control sequence.
l.58 Thus, total size is \(4288+4=\qty
```

## Questions

- **Question 1.** Since `tex_math_dollars` and `raw_tex` are enabled extensions for pandoc markdown format, it is sufficient to use `--from=markdown` instead of `--from=markdown+tex_math_dollars+raw_tex`. Is my assumption correct?
- **Question 2.** Why am I getting this error? I read [LaTeX: Support `siunitx` Commands · Issue #3587 · jgm/pandoc](https://github.com/jgm/pandoc/issues/3587), this indicate that `siunitx`  is enabled by default with Pandoc. If not do I need to enable it? If so, how can I do it?
- **Question 3.** How to get a list of default supported and enabled latex packages in pandoc (irrespective of if they are installed or not)? On [Pandoc - Pandoc User’s Guide](https://pandoc.org/MANUAL.html), there is no mention of `siunitx`.
- **Question 4.** Since `raw_tex` is not available with `commonmark_x` does it mean that we can not use the following or does it mean that enabled latex package like `siunitx` can be used freely with commonmark_x?

```
$ printf 'Thus, total size is $4288+4=\qty{4292}{kiB}$' | pandoc -s --from=commonmark_x --to latex -o a.pdf
Error producing PDF.
! Undefined control sequence.
l.58 Thus, total size is \(4288+4=\qty
```

**Comment by jgm** [comment, marked as answer]
Q1: yes
Q2: the issue you cite has to do with parsing siunitx and converting it to other formats.  The siunitx package is not included  in the default pandoc template, so math that contains macros defined by this package won't compile.  The solution is to add `\usepackage{siunitx}` either in a default template or via header-includes.
Q4: note that `raw_tex` only matters for tex outside of a math context.  Stuff in math mode just gets put in a Math element and will go through to latex verbatim.
